platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/SavedStateMetadataTests.java

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform;
+
+import static com.swirlds.common.test.RandomUtils.getRandomPrintSeed;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.system.BasicSoftwareVersion;
+import com.swirlds.common.system.SoftwareVersion;
+import com.swirlds.common.test.RandomUtils;
+import com.swirlds.platform.state.signed.SavedStateMetadata;
+import com.swirlds.platform.state.signed.SavedStateMetadataField;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+@DisplayName("SignedStateMetadata Tests")
```

Comment:
Up to you which tests (if any) you would like to add. I notice the following untested cases:
- attempting to parse string value containing newline
- file doesn't exist
- line missing colon
- whitespace is successfully stripped from around keys and values
- signing nodes are successfully sorted
- malformed long value
- malformed hash value
- long list correctly handles empty parts
- long list correctly strips whitespace surrounding values
- long list correctly handles malformed values
- newlines are serialized to `//`

Reply:
added a bunch of additional tests, all requested cases are now covered